### PR TITLE
Cleanups for #1368

### PR DESCRIPTION
**Pull request by ikoveshnikov**

### CODE REVIEW

File: tempesta_fw/fill_hpack_hdr.pl [hunk 1]
```diff
@@ -0,0 +1,963 @@
+#!/usr/bin/env perl
```
Comment (krizhanovsky) [inline tempesta_fw/fill_hpack_hdr.pl]: Commited original `hgen.c` with minor fix and coding style adjustments.

File: tempesta_fw/hpack_tbl.h.tpl [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/**
```
Comment (krizhanovsky) [inline tempesta_fw/hpack_tbl.h.tpl]: I think we don't need the template

File: tempesta_fw/hpack.c [hunk 3]
```diff
@@ -28,919 +28,7 @@
...
-	{7, -97},		/* 3: '\x9F' (159) */
-/* --- [TABLE-8: offset = 840] --- */
-	{6, 10},		/* 2: '\x0A' (10) */
-	{6, 10},		/* 2: '\x0A' (10) */
-	{6, 13},		/* 2: '\x0D' (13) */
-	{6, 13},		/* 2: '\x0D' (13) */
-	{6, 22},		/* 2: '\x16' (22) */
-	{6, 22},		/* 2: '\x16' (22) */
-	{6, 0},			/* 2: EOS */
-	{6, 0},			/* 2: EOS */
```
[the remark this answers is not in the document]
Comment (krizhanovsky) [reply tempesta_fw/hpack.c]: Please remove `hpack_tbl.h.tpl` - now we don't need it

File: Makefile [hunk 4]
```diff
@@ -74,7 +74,10 @@ obj-m	+= lib/ tempesta_db/core/ tempesta_fw/ tls/
 
 all: build
 
-build:
+generate_headers:
+	tempesta_fw/fill_hpack_hdr.pl tempesta_fw/hpack_tbl.h.tpl
```
[the remark this answers is not in the document]
Comment (krizhanovsky) [reply Makefile]: Please remove `hpack_tbl.h.tpl` - now we don't need it

File: tempesta_fw/hpack_tbl.h.tpl (hunk 2 above)
Comment (krizhanovsky) [reply tempesta_fw/hpack_tbl.h.tpl]: Please remove `hpack_tbl.h.tpl` - now we don't need it

File: tempesta_fw/hpack.c [hunk 5]
```diff
@@ -3655,6 +2742,253 @@ write_int(unsigned long index, unsigned short max, unsigned short mask,
...
+ *
+ * Since these reasons we can't encode @str on the fly or encode it while
+ * writing to skb, so allocate a new string and copy @str encoded value there.
+ * It heavily affect performance, since we have two allocations and two copy
+ * operations here. We decided to keep the code, but our core use cases shows
+ * no performance improvement opportunities.
+ *
+ * TODO: We are free to choose encoding when we're adding a new header, but
+ * we have to (?) preserve original encoding when modifying already existent
+ * headers.
```
Comment (aleksostapenko) [inline tempesta_fw/hpack.c]: According to RFC we are free to choose encoding (static/dynamic/Huffman) when modifying already existent headers too (e.g. in cases of HTTP/1.1=>HTTP/2  or HTTP/2=>HTTP/2 response proxy). Thus, this is a matter of our implementation decisions. E.g. in future implementation of HTTP/2=>HTTP/2 request forwarding it is better to leave the Huffman-encoding as is from the client request for the better performance. But yes - for the responses forwarding the Huffman-encoding has no any sense for us in context of performance (neither for added headers nor for existing ones).

Comment (ikoveshnikov) [reply tempesta_fw/hpack.c]: Thank you for clearing up, I will add this note no comment

**Pull request merged**